File: Java/Aksesoris.java
public class Aksesoris extends Petshop{
    private String jenis;
    private String bahan;
    private String warna;

    // Konstruktor default
    public Aksesoris() {
        super(); // Konstruktor Petshop
        this.jenis = "";
        this.bahan = "";
        this.warna = "";
    }

    // Konstruktor dengan parameter
    public Aksesoris(String ID, String nama, int hargaProduk, int stokProduk, String jenis, String bahan, String warna) {
        super(ID, nama, hargaProduk, stokProduk); // Konstruktor Petshop
        this.jenis = jenis;
        this.bahan = bahan;
        this.warna = warna;
    }

    // Setter
    public void set_jenis(String jenis) { this.jenis = jenis; }
    public void set_bahan(String bahan) { this.bahan = bahan; }
    public void set_warna(String warna) { this.warna = warna; }

    // Getter
    public String get_jenis() { return this.jenis; }
    public String get_bahan() { return this.bahan; }
    public String get_warna() { return this.warna; }
}

File: CPP/Aksesoris.cpp
#include <iostream>
#include <string>
#include "Petshop.cpp"

class Aksesoris : public Petshop
{
    private:
        std::string jenis;
        std::string bahan;
        std::string warna;
    
    public:
        // Konstruktor
        Aksesoris() : Petshop(){
            this->jenis = "";
            this->bahan = "";
            this->warna = "";
        }

        // Konstruktor dengan parameter
        Aksesoris(std::string ID, std::string nama, int harga_produk, int stok_produk, std::string jenis, std::string bahan, std::string warna) 
        : Petshop(ID, nama, harga_produk, stok_produk){
            this->jenis = jenis;
            this->bahan = bahan;
            this->warna = warna;
        }

        // Setter
        void set_jenis(std::string jenis) {this->jenis = jenis;}
        void set_bahan(std::string bahan) {this->bahan = bahan;}
        void set_warna(std::string warna) {this->warna = warna;}

        // Getter
        std::string get_jenis() {return this->jenis;}
        std::string get_bahan() {return this->bahan;}
        std::string get_warna() {return this->warna;}

};